tests/pthread/test_pthread_cleanup.cpp

```diff
@@ -16,95 +16,83 @@
-   pthread_cleanup_pop((int)arg);
-   pthread_exit(0);
+static void *thread_start1(void *arg) {
+  pthread_cleanup_push(cleanup_handler1, (void*)(42 + (int)arg*100));
+  pthread_cleanup_push(cleanup_handler2, (void*)(69 + (int)arg*100));
+  pthread_cleanup_pop((int)arg);
+  pthread_cleanup_pop((int)arg);
+  pthread_exit(0);
 }
 
-static void *thread_start2(void *arg)
-{
-   pthread_cleanup_push(cleanup_handler1, (void*)52);
-   pthread_cleanup_push(cleanup_handler2, (void*)79);
-   if (arg)
-      pthread_exit(0);
-   pthread_cleanup_pop(0);
-   pthread_cleanup_pop(0);
-   return 0;
+static void *thread_start2(void *arg) {
+  pthread_cleanup_push(cleanup_handler1, (void*)52);
+  pthread_cleanup_push(cleanup_handler2, (void*)79);
+  if (arg)
+    pthread_exit(0);
+  pthread_cleanup_pop(0);
+  pthread_cleanup_pop(0);
+  return 0;
 }
 
-static void *thread_start3(void *arg)
-{
-   pthread_cleanup_push(cleanup_handler1, (void*)62);
-   pthread_cleanup_push(cleanup_handler2, (void*)89);
-   for(;;)
-   {
-      pthread_testcancel();
-   }
-   pthread_cleanup_pop(0);
-   pthread_cleanup_pop(0);
-   pthread_exit(0);
+static void *thread_start3(void *arg) {
+  pthread_cleanup_push(cleanup_handler1, (void*)62);
+  pthread_cleanup_push(cleanup_handler2, (void*)89);
+  for (;;) {
+     pthread_testcancel();
+  }
+  pthread_cleanup_pop(0);
+  pthread_cleanup_pop(0);
+  pthread_exit(0);
 }
 
 pthread_t thr[4];
 
-int main()
-{
-   int result = 0;
+int main() {
+  int result = 0;
 
-   if (!emscripten_has_threading_support())
-   {
-#ifdef REPORT_RESULT
-      REPORT_RESULT(907640832);
-#endif
-      printf("Skipped: Threading is not supported.\n");
-      return 0;
-   }
+  if (!emscripten_has_threading_support()) {
+     printf("Skipped: Threading is not supported.\n");
+     return 0;
+  }
 
-   pthread_cleanup_push(cleanup_handler1, (void*)9998);
-   pthread_cleanup_push(cleanup_handler1, (void*)9999);
+  pthread_cleanup_push(cleanup_handler1, (void*)9998);
+  pthread_cleanup_push(cleanup_handler1, (void*)9999);
 
-   int s = pthread_create(&thr[0], NULL, thread_start1, (void*)0);
-   assert(s == 0);
-   pthread_join(thr[0], 0);
-   s = pthread_create(&thr[1], NULL, thread_start1, (void*)1);
-   assert(s == 0);
-   pthread_join(thr[1], 0);
-   s = pthread_create(&thr[2], NULL, thread_start2, (void*)1);
-   assert(s == 0);
-   pthread_join(thr[2], 0);
+  int s = pthread_create(&thr[0], NULL, thread_start1, (void*)0);
+  assert(s == 0);
+  pthread_join(thr[0], 0);
+  s = pthread_create(&thr[1], NULL, thread_start1, (void*)1);
+  assert(s == 0);
+  pthread_join(thr[1], 0);
+  s = pthread_create(&thr[2], NULL, thread_start2, (void*)1);
+  assert(s == 0);
+  pthread_join(thr[2], 0);
 // TODO
 //   s = pthread_create(&thr[3], NULL, thread_start3, (void*)1);
 //   assert(s == 0);
 //   s = pthread_cancel(thr[3]);
 //   assert(s == 0);
-   pthread_cleanup_pop(1);
-   EM_ASM(console.log('Cleanup state variable: ' + $0), cleanupState);
```

Comment:
This is not necessary anymore?

Reply:
You mean the use of EM_ASM over printf?  I  don't think here is any reason not to just use printf here, is simpler, and we already use printf elsewhere in this test.